### ISSUE TITLE
Integration of PhishTank API

### ISSUE THREAD

**Issue by ozh** [YOURLS/YOURLS#380]
This is a **COPY** of **Issue 380: [Integration of PhishTank API](http://code.google.com/p/yourls/issues/detail?id=380)**, filed on Google Code before the project was [moved on Github](https://github.com/ozh/google-issues-to-github).
- Submitted on 2010-06-28T02:40:41.000Z by [pawlec...@gmail.com](http://code.google.com/u/102572002598305556196/)
- Status: WontFix

Please **review** the original issue and especially its comments. **Comments here on closed issues will be ignored**. Thanks.
## Original description

```
I am personally integrating the PhishTank API into my site as a result of a bad experience, which could happen to anyone using Yourls.

A person created a phishing site and then shortened their link which then of course has my domain name attached. It was immediately reported and the link was blocked by Google and multiple sites and anti-malware programs that use the PhishTank db as a reference.

The link was of course removed immediately when Google actually contacted me about it. The link was removed within an hour.

Although the specific link was reported and not my root domain, there are sites and software that resolve to block the root domain, despite that the issue was a particular link.

As a result, my domain had become blocked by many. People trying to share links were finding their links being blocked.

This phishing site that shortened a link through my site created an aftermath of issues to be resolved, and are still being resolved. It has been 1 month now, and new issues are still being reported to me. There is a long term ripple because not all sites that use the PhishTank API, nor people with protection software, update on a regular basis.  

Overall, this was and still is damaging to my slowly building business. This could have all been prevented if I had taken the necessary means to protect my site by integrating the PhishTank API in the first place, but of course, never expected this to happen.

Integration of PhishTank's API plays a very important role in protecting one's own site. One phish site shortened and published, and it is a mess to get out of, when you find your own domain being blocked. It's a business killer.

By intgegrateing PhishTank's API, once a URL is submitted, the URL can be checked against their regularly updated list. If found on the list, then a redirect can happen instead of shortening the link, with a message and also  a link to show the actual report from PhishTank.
```
